### PR TITLE
Compare negative binomial to poisson noise

### LINKED ISSUE #222: negative binomial forwards model
- [ ] Make a forwards model with negative binomial distribution (use sensible parametrization and document it carefully)
- [ ] Make a notebook where you generate data from this, and try to learn Rt using the existing poisson based posterior. Show that as overdsipersion increases, it gets worse

### PR DESCRIPTION
**Pull request by I-Bouros**
In this pull request we have implemented equivalent classes for the branching processes in the original project, with the exception that these new classes allow for negative binomial-distributed noise.

We also include a python notebook in which we compare the effects of different values of the overdispersion parameter associated with the negative binomial noise in the data on the inferred $R_t$ trajectory (inferred under the assumption that the data is poisson, and not negative-binomial).

Closes #222

### CODE REVIEW

File: branchpro/new_models.py [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+
+    where :math:`\mu` is mean of the distribution defined as above and
+    :math:`\phi` is the overdispersion parameter associated with the negative
+    binomial noise distribution.
+
+    For the edge case, :math:`\phi = 0`, the incidence at time t becomes
+    Poisson distributed, reducing to the simple class:`BranchProModel` class.
+
+    Always apply method :meth:`set_r_profile` before calling
+    :meth:`DiffNoiseBranchProModel.simulate` for a change of R_t profile!
```
Comment (rccreswell) [inline branchpro/new_models.py:47]: Looks like you renamed it later

File: branchpro/new_models.py [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+        """
+        if not isinstance(phi, (int, float)):
+            raise TypeError(
+                'Value of overdispersion must be integer or float.')
+        if phi <= 0:
+            raise ValueError(
+                'Value of overdispersion must be must be > 0. For \
+                overdispesion = 0, please use `BranchProModel` class type.')
+
+        self._overdispesion = phi
```
Comment (rccreswell) [inline branchpro/new_models.py:107]:
```suggestion
        self._overdispersion = phi
```
etc.

File: branchpro/new_models.py [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+        # and draw samples for the incidences
+        for t in simulation_times:
+            norm_daily_mean = self._r_profile[t-1] * (
+                self._effective_no_infectives(t, incidences))
+            if norm_daily_mean != 0:
+                nu = self._overdispesion * norm_daily_mean
+                incidences[t] = nbinom.rvs(
+                    n=nu * norm_daily_mean,
+                    p=1 / (1 + nu),
+                    size=1)
```
Comment (rccreswell) [inline branchpro/new_models.py:160]: Change it to be the same as the local imported version below, which is the one I think we got working

**Review by rccreswell — CHANGES_REQUESTED** [review]
Plus do you know why the doc tests are failing

**Comment by I-Bouros** [comment]
> Plus do you know why the doc tests are failing

Probably I haven't done the docs

File: branchpro/new_models.py [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+            \frac{\Gamma(k+\frac{1}{\phi})}{k!\Gamma(\frac{1}{\phi})}
+            \Big(\frac{1}{1+\mu\phi}\Big)^{\frac{1}{\phi}}
+            \Big(\frac{\mu\phi}{1+\mu\phi}\Big)^{k}
+
+    where :math:`\mu` is mean of the distribution defined as above and
+    :math:`\phi` is the overdispersion parameter associated with the negative
+    binomial noise distribution.
+
+    For the edge case, :math:`\phi = 0`, the incidence at time t becomes
+    Poisson distributed, reducing to the simple class:`LocImpBranchProModel`
```
Comment (rccreswell) [inline branchpro/new_models.py:204]:
```suggestion
    Poisson distributed, reducing to the simple :class:`LocImpBranchProModel`
```

File: branchpro/new_models.py [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+            \frac{\Gamma(k+\frac{1}{\phi})}{k!\Gamma(\frac{1}{\phi})}
+            \Big(\frac{1}{1+\mu\phi}\Big)^{\frac{1}{\phi}}
+            \Big(\frac{\mu\phi}{1+\mu\phi}\Big)^{k}
+
+    where :math:`\mu` is mean of the distribution defined as above and
+    :math:`\phi` is the overdispersion parameter associated with the negative
+    binomial noise distribution.
+
+    For the edge case, :math:`\phi = 0`, the incidence at time t becomes
+    Poisson distributed, reducing to the simple class:`BranchProModel` class.
```
Comment (rccreswell) [inline branchpro/new_models.py:44]:
```suggestion
    Poisson distributed, reducing to the simple :class:`BranchProModel` class.
```

**Review by rccreswell — APPROVED** [review]
:+1:

**Pull request merged**